services/venice-server/src/main/java/com/linkedin/venice/listener/request/GetRouterRequest.java

```diff
@@ -51,24 +50,34 @@ public int getKeyCount() {
     return 1;
   }
 
-  public static GetRouterRequest parseGetHttpRequest(HttpRequest request, String[] requestParts) {
-    if (requestParts.length == 5) {
-      // [0]""/[1]"action"/[2]"store"/[3]"partition"/[4]"key"
-      String topicName = requestParts[2];
-      int partition = Integer.parseInt(requestParts[3]);
-      byte[] keyBytes = getKeyBytesFromUrlKeyString(requestParts[4]);
-      return new GetRouterRequest(topicName, partition, keyBytes, request);
-    } else {
+  public static GetRouterRequest parseSingleGetHttpRequest(HttpRequest request, String[] requestParts) {
+    if (requestParts.length != 5) {
       throw new VeniceException("Not a valid request for a STORAGE action: " + request.uri());
     }
+    // [0]""/[1]"action"/[2]"store"/[3]"partition"/[4]"key"
+    String topicName = requestParts[2];
+    int partition = Integer.parseInt(requestParts[3]);
+    byte[] keyBytes = getKeyBytesFromUrlKeyString(requestParts[4]);
+    boolean isRetryRequest = containRetryHeader(request);
+    boolean isStreamingRequest = StreamingUtils.isStreamingEnabled(request);
+    return new GetRouterRequest(topicName, partition, keyBytes, isRetryRequest, isStreamingRequest);
```

Comment:
I was unaware of this implicit convention. I made the change to reduce indentation which IMO makes code easy to follow, though I understand this might be subjective. I will revert the change.